common/src/main/java/com/microsoft/identity/common/internal/msafederation/google/GoogleSignInProvider.kt

```diff
@@ -0,0 +1,208 @@
+ * that handles sign-in operations with Google using credential manager in Android.
+ * The class is internal and should not be used outside of the SDK. OneAuth should use
+ * [SignInWithGoogleApi]
+ * Call [GoogleSignInProvider.create] to create an instance of GoogleSignInProvider.
+ */
+internal class GoogleSignInProvider(private val credentialManager: CredentialManager,
+                           private val parameters: SignInWithGoogleParameters,
+                           private val webClientId: String
+) : IFederatedSignInProvider {
+
+    companion object {
+        private const val TAG = "GoogleSignInProvider"
+
+        /**
+         * Creates an instance of GoogleSignInProvider.
+         *
+         * @param parameters The parameters required for signing in with Google.
+         * @param webClientId The web client ID for Google sign-in.
+         * @return A new instance of GoogleSignInProvider. Prod must use MSA client ID.
+         */
+        @JvmStatic
+        fun create(parameters: SignInWithGoogleParameters, webClientId: String): GoogleSignInProvider {
+            return GoogleSignInProvider(CredentialManager.create(parameters.activity.applicationContext), parameters, webClientId)
+        }
+    }
+
+    /**
+     * Signs in with Google using the specified parameters.
+     * if useBottomSheet is true, it will use GetGoogleIdOption based flow as
+     * suggested by google. Otherwise, it will use GetSignInWithGoogleOption based flow.
+     *
+     * @return A Result containing the SignInWithGoogleCredential on success,
+     * or an exception on failure.
+     */
+    override suspend fun signIn(): Result<SignInWithGoogleCredential> {
+        return if (parameters.useBottomSheet) {
+            signInWithGoogleBottomSheet()
+        } else {
+            signInWithGoogle()
+        }
+    }
+
+    /**
+     * Signs in with Google using a bottom sheet UI.
+     *
+     * @return A Result containing the SignInWithGoogleCredential on success, or an exception on failure.
+     */
+    private suspend fun signInWithGoogleBottomSheet(): Result<SignInWithGoogleCredential> {
+        val googleIdOption: GetGoogleIdOption = GetGoogleIdOption.Builder()
+            .setFilterByAuthorizedAccounts(false)
+            .setServerClientId(webClientId)
+            .setAutoSelectEnabled(false)
+            .setNonce(generateNonce())
+            .build()
+
+        return getCredential(googleIdOption)
+    }
+
+    /**
+     * Signs in with Google using a standard UI.
+     *
+     * @return A Result containing the SignInWithGoogleCredential on success, or an exception on failure.
+     */
+    private suspend fun signInWithGoogle(): Result<SignInWithGoogleCredential> {
+        val signInWithGoogleOption = GetSignInWithGoogleOption.Builder(webClientId)
+            .setNonce(generateNonce())
+            .build()
+
+        return getCredential(signInWithGoogleOption)
+    }
+
+    /**
+     * Signs out from Google for this app.
```

Comment:
Is it really only from this app? Or would this result in global signout?

Reply:
only app.